pos_self_service_weight_base/static/src/js/ChromeWidgets/HeaderButton.esm.js

```diff
@@ -0,0 +1,39 @@
+/** @odoo-module alias=pos_self_service_weight_base.HeaderButton **/
+// SPDX-FileCopyrightText: 2024 Coop IT Easy SC
+//
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+import HeaderButton from "point_of_sale.HeaderButton";
+import Registries from "point_of_sale.Registries";
+import {isConnectionError} from "point_of_sale.utils";
+
+const SelfServiceHeaderButton = (HeaderButton) =>
+    class extends HeaderButton {
+        async onClick() {
+            if (!this.env.pos.config.is_self_service_weight_point) {
+                return super.onClick();
+            }
+            try {
+                const info = await this.env.pos.getClosePosInfo();
+                this.showPopup("CloseSelfServicePopup", {info: info, keepBehind: true});
+            } catch (e) {
+                if (isConnectionError(e)) {
+                    this.showPopup("OfflineErrorPopup", {
+                        title: this.env._t("Network Error"),
+                        body: this.env._t(
+                            "Please check your internet connection and try again."
+                        ),
+                    });
```

Comment:
why should an internet connection error prevent to close the weighing station? ok, i just saw that this is a copy of the `HeaderButton.onClick()` method with the `showPopup()` argument changed. isn’t there a better way? why not override the `showPopup()` method instead? if this is really needed, a comment should explain that this is a copy and what has changed.